src/passes/OptimizeInstructions.cpp

```diff
@@ -2891,6 +2891,12 @@ struct OptimizeInstructions
             }
           }
         }
+        if (unary->op == EqZInt32 || unary->op == EqZInt64) {
+          if (auto* c = getFallthrough(unary->value)->dynCast<Const>()) {
```

Comment:
Hmm, I suspect this is not safe: getFallthrough can return nullptr, and dynCast assumes the pointer is not null (it reads the type). The fuzzer should catch stuff like this.

Instead, this could be
```
if (auto* fallthrough = getFallthrough(unary->value)) {
  if (auto* c = fallthrough->dynCast<Const>()) {
```

Reply:
Oh wait, I misremembered getFallthrough: it returns the node itself, if there is no value falling through it. So no change is needed here. Sorry.